cdc/owner/schema.go

```diff
@@ -0,0 +1,158 @@
+// Copyright 2021 PingCAP, Inc.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package owner
+
+import (
+	"github.com/pingcap/errors"
+	"github.com/pingcap/log"
+	timodel "github.com/pingcap/parser/model"
+	"github.com/pingcap/ticdc/cdc/entry"
+	"github.com/pingcap/ticdc/cdc/kv"
+	"github.com/pingcap/ticdc/cdc/model"
+	"github.com/pingcap/ticdc/pkg/config"
+	"github.com/pingcap/ticdc/pkg/cyclic/mark"
+	"github.com/pingcap/ticdc/pkg/filter"
+	tidbkv "github.com/pingcap/tidb/kv"
+	timeta "github.com/pingcap/tidb/meta"
+	"go.uber.org/zap"
+)
+
+type schemaWrap4Owner struct {
+	schemaSnapshot *entry.SingleSchemaSnapshot
+	filter         *filter.Filter
+	config         *config.ReplicaConfig
+
+	allPhysicalTablesCache []model.TableID
+}
+
+func newSchemaWrap4Owner(kvStorage tidbkv.Storage, startTs model.Ts, config *config.ReplicaConfig) (*schemaWrap4Owner, error) {
+	var meta *timeta.Meta
+	if kvStorage != nil {
+		var err error
+		meta, err = kv.GetSnapshotMeta(kvStorage, startTs)
+		if err != nil {
+			return nil, errors.Trace(err)
+		}
+	}
+	schemaSnap, err := entry.NewSingleSchemaSnapshotFromMeta(meta, startTs, config.ForceReplicate)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	f, err := filter.NewFilter(config)
+	if err != nil {
+		return nil, errors.Trace(err)
+	}
+	return &schemaWrap4Owner{
+		schemaSnapshot: schemaSnap,
+		filter:         f,
+		config:         config,
+	}, nil
+}
+
+// AllPhysicalTables returns the table IDs of all tables and partition tables.
+func (s *schemaWrap4Owner) AllPhysicalTables() []model.TableID {
+	if s.allPhysicalTablesCache != nil {
+		return s.allPhysicalTablesCache
+	}
+	tables := s.schemaSnapshot.CloneTables()
```

Comment:
Can we expose `tables` from `schemaSnapshot` directly, there is no need to copy it.
Besides, `CloneTables` only returns table IDs from `s.tables`, it doesn't count the `partitionTable`, is this a bug?